### PR TITLE
Replace Pact Service Module Cache with Checkpointer Cache

### PR DESCRIPTION
**Pull request by ak3n**
This is a work in progress PR exploring the direction of replacing Pact Service's module cache with Checkpointer's cache.

### CODE REVIEW

File: chainweb.cabal [hunk 1]
```diff
@@ -425,7 +425,7 @@ library
 -- -------------------------------------------------------------------------- --
 
 test-suite chainweb-tests
-    import: warning-flags, debugging-flags
+    import: debugging-flags
```
Comment (chessai) [inline chainweb.cabal:428]: is this warning removal just for local dev? If so, I believe you can just add a cabal.project.local with the appropriate settings.

File: src/Chainweb/Pact/Backend/DbCache.hs [hunk 2]
```diff
@@ -193,12 +230,60 @@ cacheStats mc = object
...
+keysDbCache :: DbCache PersistModuleData -> [CacheAddress]
+keysDbCache DbCache{_dcStore} = HM.keys _dcStore
+
 -- -------------------------------------------------------------------------- --
 -- Internal
 
 mkAddress :: Utf8 -> ByteString -> CacheAddress
 mkAddress (Utf8 key) rowdata = CacheAddress . BS.toShort $
-    key <> ":" <> BA.convert (C.hash @_ @SHA512t_256 rowdata)
+    "hashed:" <> key <> ":" <> BA.convert (C.hash @_ @SHA512t_256 rowdata)
```
Comment (chessai) [inline src/Chainweb/Pact/Backend/DbCache.hs:286]: why this change?

File: chainweb.cabal (hunk 1 above)
Comment (ak3n) [reply chainweb.cabal:428]: Yep, that's what I do for local development, too lazy to explore `cabal.project.local`

File: src/Chainweb/Pact/Backend/DbCache.hs (hunk 2 above)
Comment (ak3n) [reply src/Chainweb/Pact/Backend/DbCache.hs:286]: We have to filter out these keys as they consist of non-unicode hashes while `ModuleName` type is a `Text`, and they are very specific.

**Comment by ak3n** [comment]
Closing this PR as it we've figured out a better solution that will follow after shallow nodes support.
